src/views/toolbar/data-toolbar-directive.js

```diff
@@ -0,0 +1,248 @@
+      };
+
+      var filterChange = function (filters) {
+      $scope.filtersText = "";
+        filters.forEach(function (filter) {
+          $scope.filtersText += filter.title + " : " + filter.value + "\n";
+        });
+        applyFilters(filters);
+        $scope.toolbarConfig.filterConfig.resultsCount = $scope.items.length;
+      };
+
+      $scope.filterConfig = {
+        fields: [
+          {
+            id: 'name',
+            title:  'Name',
+            placeholder: 'Filter by Name',
+            filterType: 'text'
+          },
+          {
+            id: 'address',
+            title:  'Address',
+            placeholder: 'Filter by Address',
+            filterType: 'text'
+          },
+          {
+            id: 'birthMonth',
+            title:  'Birth Month',
+            placeholder: 'Filter by Birth Month',
+            filterType: 'select',
+            filterValues: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
+          }
+        ],
+        resultsCount: $scope.items.length,
+        appliedFilters: [],
+        onFilterChange: filterChange
+      };
+
+      var viewSelected = function(viewId) {
+        $scope.viewType = viewId
+      };
+
+      $scope.viewsConfig = {
+        views: [pfViewUtils.getListView(), pfViewUtils.getTilesView()],
+        onViewSelect: viewSelected
+      };
+      $scope.viewsConfig.currentView = $scope.viewsConfig.views[0].id;
+      $scope.viewType = $scope.viewsConfig.currentView;
+
+      $scope.toolbarConfig = {
+        viewsConfig: $scope.viewsConfig,
+        filterConfig: $scope.filterConfig
+      };
+
+      $scope.listConfig = {
+        selectionMatchProp: 'name',
+        checkDisabled: false
+      };
+    }
+  ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.views').directive('pfDataToolbar',
+  function () {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      replace: true,
+      transclude: false,
+      templateUrl: 'views/toolbar/data-toolbar.html',
+      controller: function ($scope) {
+        $scope.viewSelected = function (viewId) {
+          $scope.config.viewsConfig.currentView = viewId;
+          if ($scope.config.viewsConfig.onViewSelect && !$scope.checkViewDisabled(viewId)) {
+            $scope.config.viewsConfig.onViewSelect(viewId);
+          }
+        };
+
+        $scope.isViewSelected = function (viewId) {
+          return $scope.config.viewsConfig && ($scope.config.viewsConfig.currentView === viewId);
+        };
+
+        $scope.checkViewDisabled = function (view) {
+          return $scope.config.viewsConfig.checkViewDisabled && $scope.config.viewsConfig.checkViewDisabled(view);
+        };
+      },
+      link: function (scope, element, attrs) {
+        scope.$watch('config', function () {
+          if (scope.config && scope.config.viewsConfig && scope.config.viewsConfig.views) {
```

Comment:
No, watching and observing have to be done post compile (ie in the link function)